### PR TITLE
fix: add new modal to allow selecting which wallet to use

### PR DESCRIPTION
**Pull request by janniks**
> This PR was published to npm with the alpha versions:
> - connect `npm install @stacks/connect@7.4.3-alpha.edd9f52.0 --save-exact`
> - connect-react `npm install @stacks/connect-react@22.2.3-alpha.edd9f52.0 --save-exact`
> - connect-ui `npm install @stacks/connect-ui@6.1.4-alpha.edd9f52.0 --save-exact`

- A potential solution for #325 
- Allows for selecting available providers, eliminating conflicts

Screenshot 2023-11-15 at 15 26 41

### CODE REVIEW

**Comment by eugeniadigon** [comment]
Here's the revised UI for this modal: https://www.figma.com/file/taNNRIyF5NvW2a2KmL1MB8/Connect-UI?type=design&node-id=0%3A1&mode=design&t=Lm6PwiJjB1L3ndLZ-1

Let me know if you have any comments!

browser-1
browser-2
browser

**Comment by janniks** [comment]
@eugeniadigon thanks 🙏 
Screenshot 2023-11-28 at 14 51 01
Updated ✅ _I made some minor changes_

**Comment by beguene** [comment]
@janniks I added this to the hiro platform to test.
The popup display the available wallets correctly

Screenshot 2023-11-29 at 15 31 23

Screenshot 2023-11-29 at 15 33 18

But when I installed Xverse and reloaded my page, then clicked on 'Connect', then in the popup I select Xverse, the Leather Wallet pops up instead

https://github.com/hirosystems/connect/assets/1501454/32b94998-697e-43e0-b396-2c1e4385c9c2

File: packages/connect/src/ui.ts [hunk 1]
```diff
@@ -1,34 +1,127 @@
...
-  provider: StacksProvider = getStacksProvider()
-) => {
-  if (provider) {
-    void authenticate(authOptions, provider);
-    return;
-  }
-
-  if (typeof window !== undefined) {
+
+export const DEFAULT_PROVIDERS: WebBTCProvider[] = [
```
Comment (beguene) [inline packages/connect/src/ui.ts:32]: @janniks maybe this DEFAULT_PROVIDERS could be in its own file, to make it easier for someone not familiar with the code to add or update a provider.

Comment (beguene) [inline packages/connect/src/ui.ts:32]: @janniks maybe this DEFAULT_PROVIDERS could be in its own file, to make it easier for someone not familiar with the code to add or update a provider.

Comment (janniks) [reply packages/connect/src/ui.ts]: Good call!

**Comment by janniks** [comment]
> But when I installed Xverse and reloaded my page, then clicked on 'Connect', then in the popup I select Xverse, the Leather Wallet pops up instead

Thanks @beguene I forgot to add an important handling of providers 🫡 Should be fixed now

**Comment by beguene** [comment]
@janniks I installed the latest version

```
    "@stacks/connect": "7.4.3-alpha.92feb0f.0",
    "@stacks/connect-react": "22.2.3-alpha.92feb0f.0",
```

But I get this
image
I also tried with browser cache disabled.

**Comment by janniks** [comment]
Try adding `npm install @stacks/connect-ui@6.1.4-alpha.92feb0f.0 --save-exact` -- alpha versions are often not used exactly for dependencies, because npm doesn't know how to follow the semver for commit SHAs.

I just tried on the starters repo and it seems to work 👍

**Review by beguene — APPROVED** [review]
Thanks @janniks, `--save-exact` did the trick.

Code looks good and everything works as intended.

 :shipit:

**Comment by janniks** [comment]
Wonderful thanks for the review 🙇

**Pull request merged**